src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright 2024 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard.agents',
+  name: 'ReturnToLaunchPointAgent',
+  flags: ['web'],
+  documentation: `
+  Handle route changes on wizard close
+  `,
+
+  imports: [
+    'currentMenu',
+    'ctrl',
+    'isIframe',
+    'pushDefaultMenu',
+    'stack'
+  ],
+
+  implements: [
+    'foam.core.ContextAgent'
+  ],
+
+  methods: [
+    async function execute() {
+      if ( this.isIframe() ) return;
+      if ( this.currentMenu.id === this.ctrl.route ) {
+        if ( this.stack.pos >= 0 && window.history.length > 1 )
```

Comment:
stack is no longer a reliable way to check the place in the app
```suggestion
        if ( window.history.length > 1 )
```

Reply:
thanks, updated